python312: Unbreak build with tkinter option

outpaddling opened the pull request:
Add -I${PREFIX}/include/tcl/generic

outpaddling committed:
python312: Unbreak build with tkinter option

Add -I${PREFIX}/include/tcl/generic

jperkin commented on lang/python312/options.mk:
```diff
@@ -18,8 +18,9 @@ PLIST.dtrace=		yes
 PLIST.tkinter=	yes
 .include "../../lang/tcl/buildlink3.mk"
 .include "../../x11/tk/buildlink3.mk"
+CFLAGS+=	-I${PREFIX}/include/tcl/generic
```
`BUILDLINK_PREFIX.tcl`, but also this should maybe be added to `BUILDLINK_INCDIRS.tcl` if it's a common problem.

outpaddling committed:
python312: Use bl3 vars to specify TCL includes

Also add BUILDLINK_INCDIRS.tcl to tcl bl3
There are a few other instances where this could be useful
